test/storage-luatest/storage_1_1_test.lua

```diff
@@ -0,0 +1,160 @@
+local t = require('luatest')
+local vtest = require('test.luatest_helpers.vtest')
+local vutil = require('vshard.util')
+local wait_timeout = 120
+
+local group_config = {{engine = 'memtx'}, {engine = 'vinyl'}}
+
+if vutil.feature.memtx_mvcc then
+    table.insert(group_config, {
+        engine = 'memtx', memtx_use_mvcc_engine = true
+    })
+    table.insert(group_config, {
+        engine = 'vinyl', memtx_use_mvcc_engine = true
+    })
+end
+
+local test_group = t.group('storage', group_config)
+
+local cfg_template = {
+    sharding = {
+        {
+            replicas = {
+                replica_1_a = {
+                    master = true,
+                },
+            },
+        },
+        {
+            replicas = {
+                replica_2_a = {
+                    master = true,
+                },
+            },
+        },
+    },
+    bucket_count = 10
+}
+local cluster_cfg
+
+test_group.before_all(function(g)
+    cfg_template.memtx_use_mvcc_engine = g.params.memtx_use_mvcc_engine
+    cluster_cfg = vtest.config_new(cfg_template)
+
+    vtest.storage_new(g, cluster_cfg)
+    vtest.storage_bootstrap(g, cluster_cfg)
+    vtest.storage_rebalancer_disable(g)
+end)
+
+test_group.after_all(function(g)
+    g.cluster:drop()
+end)
+
+--
+-- Test how bucket_send preserves tuple field types over the network (gh-327).
+--
+test_group.test_bucket_send_field_types = function(g)
+    -- Make a space with all the field types whose msgpack representation is
+    -- nontrivial.
+    local _, err = vtest.storage_exec_each_master(g, function(engine)
+        local format = {
+            {'id', 'unsigned'},
+            {'bid', 'unsigned'},
+        }
+        local i = 1
+        local tuple = {i, box.NULL}
+        -- Decimal appeared earlier, but this is when it can be used as a field
+        -- type. Same logic with other types.
+        if ivutil.version_is_at_least(2, 3, 0, nil, 0, 0) then
```

Comment:
A bit useless check because msgpack.object is available since 2.10 anyway.

Reply:
I want the tests work since 1.10. Decimal appeared before 2.10.

Reply:
Ok